### PR TITLE
Add tater guidebooks

### PR DESCRIPTION
**Pull request by haykam821**
This pull request adds tater guidebooks, which can be used to record and display every block position each tater is placed at within a world.

Tater guidebook UI

### CODE REVIEW

File: src/main/java/xyz/nucleoid/extras/lobby/item/TaterGuidebookItem.java [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+        setTaterPositions(stack, taterPositions);
+
+        player.getItemCooldownManager().set(stack.getItem(), RECORD_COOLDOWN);
+
+        int difference = taterPositions.size() - initialCount;
+        player.sendMessage(Text.translatable("text.nucleoid_extras.tater_guidebook.recorded", difference), true);
+    }
+
+    private static void recordChunk(Chunk chunk, BlockPos.Mutable pos, SetMultimap<Item, BlockPos> taterPositions) {
+        for (int z = 0; z < 16; z++) {
```
Comment (Gegy) [inline src/main/java/xyz/nucleoid/extras/lobby/item/TaterGuidebookItem.java]: `Chunk.forEachBlockMatchingPredicate`? Should be a lot more efficient, as it checks palette before doing a linear scan 🙂

File: src/main/java/xyz/nucleoid/extras/lobby/item/TaterGuidebookItem.java [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+    public TaterGuidebookItem(Settings settings) {
+        super(settings);
+    }
+
+    @Override
+    public TypedActionResult<ItemStack> use(World world, PlayerEntity user, Hand hand) {
+        var stack = user.getStackInHand(hand);
+
+        if (!world.isClient() && user.isCreativeLevelTwoOp()) {
```
Comment (Gegy) [inline src/main/java/xyz/nucleoid/extras/lobby/item/TaterGuidebookItem.java]: Hmm, should the guidebook only be usable by ops? (is it meant to be used by players? at least, viewing it)

File: src/main/java/xyz/nucleoid/extras/lobby/item/TaterGuidebookItem.java [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+
+                var item = Registries.ITEM.get(id);
+                if (item == null) continue;
+
+                var list = nbt.getList(key, NbtElement.COMPOUND_TYPE);
+                if (list == null) continue;
+
+                for (int index = 0; index < list.size(); index++) {
+                    var pos = list.getCompound(index);
+                    if (pos == null) continue;
```
Comment (Gegy) [inline src/main/java/xyz/nucleoid/extras/lobby/item/TaterGuidebookItem.java]: These NBT accessors won't ever return null

**Review by Gegy — COMMENTED** [review]
Looks good, just some small things - sorry for taking so long!

File: src/main/java/xyz/nucleoid/extras/lobby/item/TaterGuidebookItem.java (hunk 3 above)
Comment (haykam821) [reply src/main/java/xyz/nucleoid/extras/lobby/item/TaterGuidebookItem.java]: Fixed in 4934c9db1c9b0e21864a45f0d8bd9d3af30f0ceb

File: src/main/java/xyz/nucleoid/extras/lobby/item/TaterGuidebookItem.java (hunk 1 above)
Comment (haykam821) [reply src/main/java/xyz/nucleoid/extras/lobby/item/TaterGuidebookItem.java]: Fixed in 1f5a41f5bcbea791606ba84a92af6cdda44bf9ff

**Review by Gegy — APPROVED** [review]

**Pull request merged**